sdk/core/azure-core/azure/core/settings.py

```diff
@@ -72,14 +72,14 @@ def convert_bool(value):
     :raises ValueError: If conversion to bool fails
 
     """
-    if value in (True, False):
-        return value  # type: ignore
-
-    val = value.lower()  # type: ignore
-    if val in ["yes", "1", "on", "true", "True"]:
-        return True
-    if val in ["no", "0", "off", "false", "False"]:
-        return False
+    if isinstance(value, bool):
+        return value
+    if isinstance(value, str):
```

Comment:
This is a good scenario for a cast, instead of using a `isinstance`. Here you tell mypy with a cast "I took care of the boolean part of the `Union[str, bool]`, now I know this is a string"